### DISCUSSION TITLE
Does Godot team have any plan to make GDScript compilable to machine binary code, like in case of .NET Native in C#?

### DISCUSSION THREAD

**Discussion by Foxchandaisuki** [godotengine/godot-proposals#9454] [Engine Core]
Does Godot team have any plan to make GDScript compilable to machine binary code, like in case of .NET Native in C#?

Of course I know GDScript is an interpreted language, but it doesn’t mean that it cannot be AOT(Ahead-of-Time) compiled.

Dart language developed by Google already supports AOT compilation and JIT(Just-in-Time) compilation along with pure interpretation.

If GDScript supports Native AOT compilation feature, it will make games and programs written in GDScript more secure and faster.

**Comment by AThousandShips** [comment]
Thank you for your proposal, please see:
* https://github.com/godotengine/godot-proposals/issues/3069
* https://github.com/godotengine/godot-proposals/issues/6031
* https://github.com/godotengine/godot-proposals/issues/8605
* https://github.com/godotengine/godot-proposals/issues/3370
